### PR TITLE
fix: cowswap trades displaying affiliateBps as undefined

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Makes `affiliateBps` and `potentialAffiliateBps` required regardless of wallet support - this is displayed at view-layer, and we want to ensure these are defined, even if they're zero'd out.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/6160

Low, theoretically isolated to CoWSwapper, though `affiliateBps` and `potentialAffiliateBps` having been made undefined requires a paranoia check on affiliateBps (with and without FOX discounts on) on other swappers as a paranoia check.
When doing so, don't forget that we currently aggregate FOX power over wallets so you'll either want to bust your cache for both develop and this diff (or `release` if testing it there), *or* use a fresh browser profile with the same wallet used in both.

It is worth noting that this behavior has been introduced in a types refactor and wasn't the original intended behavior, so we are realistically safe here, as this reverts back to the indended behavior from the wrongly-introduced type change.

> What protocols, transaction types or contract interactions might be affected by this PR?

None - CoW doesn't support affiliateBps and as such, `AppData` doesn't support it so there's no affiliateBps being used in the signed message
 
## Testing

- CowSwap no longer shows undefined bps at confirm step
- affiliate bps are still happy for other swappers

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #6163** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6163?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/6163?utm_source=stack-comment).

File: src/lib/swapper/swappers/CowSwapper/endpoints.ts [hunk 1]
```diff
@@ -45,6 +45,8 @@ export const cowApi: SwapperApi = {
   ): Promise<Result<TradeQuote[], SwapErrorRight>> => {
     const tradeQuoteResult = await getCowSwapTradeQuote(input as GetEvmTradeQuoteInput)
 
+    console.log({ input, tradeQuoteResult })
+
```
Comment (kaladinlight) [inline src/lib/swapper/swappers/CowSwapper/endpoints.ts:49]:
```suggestion
```

File: packages/swapper/src/types.ts [hunk 2]
```diff
@@ -140,8 +140,8 @@ type TradeQuoteBase = {
   rate: string // top-level rate for all steps (i.e. output amount / input amount)
   receiveAddress: string
   receiveAccountNumber?: number
-  potentialAffiliateBps: string | undefined // undefined if affiliate fees aren't supported by the swapper
-  affiliateBps: string | undefined // undefined if affiliate fees aren't supported by the swapper
+  potentialAffiliateBps: string | undefined // even if the swapper does not support affiliateBps, we need to zero-them out or view-layer will be borked
+  affiliateBps: string | undefined // // even if the swapper does not support affiliateBps, we need to zero-them out or view-layer will be borked
```
Comment (kaladinlight) [inline packages/swapper/src/types.ts:144]: make required and i think the comment in cowswap quote is sufficient imo
```suggestion
  potentialAffiliateBps: string
  affiliateBps: string
```

File: src/state/slices/tradeQuoteSlice/selectors.ts [hunk 3]
```diff
@@ -538,6 +538,7 @@ export const selectQuoteSellAmountUserCurrency = createSelector(
 
 export const selectActiveQuoteAffiliateBps: Selector<ReduxState, string | undefined> =
   createSelector(selectActiveQuote, activeQuote => {
+    console.log({ activeQuote })
```
Comment (kaladinlight) [inline src/state/slices/tradeQuoteSlice/selectors.ts:541]:
```suggestion
```

File: src/lib/swapper/swappers/utils/test-data/setupSwapQuote.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ export const setupQuote = () => {
     id: 'foobar',
     receiveAddress: '0x1234',
     affiliateBps: undefined,
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/lib/swapper/swappers/utils/test-data/setupSwapQuote.ts]: 6252df3c10

File: packages/swapper/src/types.ts (hunk 2 above)
Comment (gomesalexandre) [reply packages/swapper/src/types.ts]: absolutely - 6252df3c10

File: packages/swapper/src/types.ts [hunk 5]
```diff
@@ -140,8 +140,8 @@ type TradeQuoteBase = {
   rate: string // top-level rate for all steps (i.e. output amount / input amount)
   receiveAddress: string
   receiveAccountNumber?: number
-  potentialAffiliateBps: string | undefined // undefined if affiliate fees aren't supported by the swapper
-  affiliateBps: string | undefined // undefined if affiliate fees aren't supported by the swapper
+  potentialAffiliateBps: string // even if the swapper does not support affiliateBps, we need to zero-them out or view-layer will be borked
+  affiliateBps: string // // even if the swapper does not support affiliateBps, we need to zero-them out or view-layer will be borked
```
Comment (kaladinlight) [inline packages/swapper/src/types.ts:144]:
```suggestion
  affiliateBps: string // even if the swapper does not support affiliateBps, we need to zero-them out or view-layer will be borked
```

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**